src/vsr/superblock.zig

```diff
@@ -885,6 +886,22 @@ pub fn SuperBlockType(comptime Storage: type) type {
                 superblock.staging.vsr_state.checkpoint.commit_min_checksum);
             assert(update.sync_op_min <= update.sync_op_max);
 
+            const free_set_reference = superblock.free_set_encoded.checkpoint_reference();
+
+            var storage_size = data_file_size_min;
+            if (superblock.free_set.highest_address_acquired()) |address| {
+                storage_size += address * constants.block_size;
+                assert(free_set_reference.head_address != 0);
+                assert(!superblock.free_set.is_free(free_set_reference.head_address));
```

Comment:
Why is this non-free? Doesn't `FreeSetEncoded.checkpoint` free these before it is done? (`set.grid.?.superblock.free_set.release(address);`).

Aha, we release it, but it is only staged, and won't actually be freed until the _next_ checkpoint!
Maybe add an `assert(set.count_released() > 0);` (and `== 0` in the other branch) to emphasize this.

Reply:
Will check for `superblock.free_set.is_released` directly!